### PR TITLE
I've added some detection for specific topics of concern.

### PR DESCRIPTION
**Pull request by markiragoldberg**
I don't think the regexes I'm using are ideal, but the structure is in.

### CODE REVIEW

File: mhapsite/mhap/models.py [hunk 1]
```diff
@@ -15,6 +15,11 @@
 import requests, json, os
 MSFT_COGSERV_KEY = os.environ.get('MSFT_COGSERV_KEY')
 
+# for keyword searches for important topics
+import re
+depression_patterns = re.compile(r'(depression)|(depressed)|(hopeless)|(worthless)', flags=re.I)
+suicide_patterns = re.compile(r'(suicid)|(kill me)|(kill myself)|(I was dead)', flags=re.I)
```
Comment (akohli96) [inline mhapsite/mhap/models.py:21]: Mark this looks good for now. We could further encapsulate it in another python file and import that too.

File: mhapsite/mhap/models.py [hunk 2]
```diff
@@ -124,9 +131,17 @@ def evaluate_sentiment(instance):
...
+    if(patterns.search(str(instance.title) + ". " + str(instance.content))):
+        return True
+    else:
+        return False
 
 def pre_save_post_receiver(sender, instance, *args, **kwargs):
     instance.slug = create_slug(instance)
     instance.sentiment = evaluate_sentiment(instance)
+    instance.seems_depressed = contains_patterns(instance, depression_patterns)
+    instance.seems_suicidal = contains_patterns(instance, suicide_patterns)
```
Comment (akohli96) [inline mhapsite/mhap/models.py:145]: Good consistency

File: mhapsite/mhap/views.py [hunk 3]
```diff
@@ -59,13 +59,22 @@ def post_detail(request, slug=None):
...
     print "secret: " + str(instance.secret)
     print "current user: " + str(user_prof)
     print "blog user: " + str(instance.user_id)
     if instance.secret and (instance.user_id != user_prof):
         raise Http404
     print instance 
     print request.user
+    
+    if instance.seems_suicidal:
+        messages.info(request, "Please don't kill yourself.")
```
Comment (akohli96) [inline mhapsite/mhap/views.py]: This message seems abrupt. Please change it

**Review by akohli96 — APPROVED** [review]

**Review by perpetual-pelican — APPROVED** [review]
This seems like a good starting point for mental health suggestions

**Pull request merged**